src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2049,7 +2049,7 @@ public function getRouteKey()
      */
     public function getRouteKeyName()
     {
-        return $this->getKeyName();
+        return array_get(config('database'), 'route_key_name', $this->getKeyName());
```

Comment:
What would be the right approach to go about this than? Can't think of any other way to load the config or to set this global in a different way that wouldn't require the config.